lib/utils/docs_parser/parsers/parameters_parser.dart

```diff
@@ -0,0 +1,88 @@
+import 'dart:io';
+
+import 'package:analyzer/dart/analysis/features.dart';
+import 'package:analyzer/dart/analysis/utilities.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:collection/collection.dart';
+import 'package:path/path.dart';
+import 'package:solid_lints/utils/docs_parser/models/parameter_doc.dart';
+import 'package:solid_lints/utils/docs_parser/parser_utils.dart';
+
+///
+class ParametersParser {
+  static const _parametersDir = 'models';
+  static const _parametersSuffix = 'parameters';
+
+  /// Directory containing rule file
+  final Directory ruleDirectory;
+
+  ///
+  const ParametersParser({required this.ruleDirectory});
+
+  ///
+  List<ParameterDoc> parse() {
+    final parametersPath = _getParametersFilePath();
+    if (parametersPath == null) return [];
+
+    final ast = parseFile(
+      path: parametersPath,
+      featureSet: FeatureSet.latestLanguageVersion(),
+    );
+
+    final parameterDocs = <ParameterDoc>[];
+    for (final declaration in ast.unit.declarations) {
+      if (declaration is ClassDeclaration) {
+        parameterDocs.addAll(_parseParametersDocs(declaration));
+        if (parameterDocs.isNotEmpty) break;
+      }
+    }
+
+    return parameterDocs;
+  }
```

Comment:
Same as the first comment. May need `package:collection` if it's not there yet: 
```suggestion
    final parameterDocs = ast.unit.declarations
      .whereType<ClassDeclaration>()
      .map(_parseParametersDocs)
      .firstWhereOrNull((docs) => docs.isNotEmpty)
      .expand((e) => e);

    return parameterDocs;
  }
```